### PR TITLE
- [gst-plugins-good] 1.19.1

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **gst-plugins-good/1.19.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by daravi** [comment]
@SSE4 Hi just wondering if this fell of the radar by accident or if it is blocked by something else?

**Comment by SSE4** [comment]
> @SSE4 Hi just wondering if this fell of the radar by accident or if it is blocked by something else?

yes, it's definitely on my Radar. I had long x-mas holidays, but I hope tomorrow I'll take an another look.

**Comment by SSE4** [comment]
> @SSE4 Hi just wondering if this fell of the radar by accident or if it is blocked by something else?

for the record, PRs to follow up: #8378 and #8800

File: recipes/gst-plugins-good/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                   "videocrop",
+                   "videofilter",
+                   "videomixer",
+                   "wavenc",
+                   "wavparse",
+                   "y4menc"]
+
+        gst_plugin_path = os.path.join(self.package_folder, "lib", "gstreamer-1.0")
+        if self.options.shared:
+            self.output.info("Appending GST_PLUGIN_PATH env var : %s" % gst_plugin_path)
```
Comment (SSE4) [inline recipes/gst-plugins-good/all/conanfile.py:197]:
```suggestion
            self.output.info("Appending GST_PLUGIN_PATH env var : %s" % gst_plugin_path)
            self.cpp_info.bindirs.append(gst_plugin_path)
```

File: recipes/gst-plugins-good/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            self.cpp_info.bindirs.append(gst_plugin_path)
+            self.env_info.GST_PLUGIN_PATH.append(gst_plugin_path)
```
Comment (SpaceIm) [inline recipes/gst-plugins-good/all/conanfile.py:199]:
```suggestion
            self.cpp_info.bindirs.append(gst_plugin_path)
            self.runenv_info.GST_PLUGIN_PATH.append(gst_plugin_path)
            self.env_info.GST_PLUGIN_PATH.append(gst_plugin_path)
```

to support `VirtualRunEnv`

**Comment by SSE4** [comment]
weird, I cannot reproduce failure locally with the same profile 🤔 perhaps, I'll try adding some additional logging. @daravi or maybe you can take a look if you have a time.

File: recipes/gst-plugins-good/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
```
Comment (SSE4) [inline recipes/gst-plugins-good/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
            os.environ["GST_DEBUG"] = "7" # TRACE
```
some debugging...
https://gstreamer.freedesktop.org/documentation/tutorials/basic/debugging-tools.html?gi-language=c#the-debug-log

**Comment by SSE4** [comment]
don't see much useful info...
```
23118:23f10 @ 818593484 - LdrpFindDllActivationContext - INFO: Probing for the manifest of DLL "C:\J\w\BuildSingleReference@2\.conan\data\gst-plugins-good\1.19.1\_\_\package\3bf60f0613e70e2876c6a5db6e066327b5b39e9c\lib\gstreamer-1.0\gstwavparse.dll" failed with status 0xc0000089
...
23118:23f10 @ 818593484 - LdrpInitializeNode - INFO: Calling init routine 00007FF941908CE4 for DLL "C:\J\w\BuildSingleReference@2\.conan\data\gst-plugins-good\1.19.1\_\_\package\3bf60f0613e70e2876c6a5db6e066327b5b39e9c\lib\gstreamer-1.0\gstwavparse.dll"
...
23118:23f10 @ 818593484 - LdrpInitializeNode - ERROR: Init routine 00007FF941908CE4 for DLL "C:\J\w\BuildSingleReference@2\.conan\data\gst-plugins-good\1.19.1\_\_\package\3bf60f0613e70e2876c6a5db6e066327b5b39e9c\lib\gstreamer-1.0\gstwavparse.dll" failed during DLL_PROCESS_ATTACH
23118:23f10 @ 818593484 - LdrpProcessDetachNode - INFO: Uninitializing DLL "C:\J\w\BuildSingleReference@2\.conan\data\gst-plugins-good\1.19.1\_\_\package\3bf60f0613e70e2876c6a5db6e066327b5b39e9c\lib\gstreamer-1.0\gstwavparse.dll" (Init routine: 00007FF941908CE4)
23118:23f10 @ 818593484 - LdrpUnloadNode - INFO: Unmapping DLL "C:\J\w\BuildSingleReference@2\.conan\data\gst-plugins-good\1.19.1\_\_\package\3bf60f0613e70e2876c6a5db6e066327b5b39e9c\lib\gstreamer-1.0\gstwavparse.dll"
23118:23f10 @ 818593484 - LdrpLoadDllInternal - RETURN: Status: 0xc0000142
23118:23f10 @ 818593484 - LdrLoadDll - RETURN: Status: 0xc0000142
```

**0xc0000089** is `STATUS_RESOURCE_DATA_NOT_FOUND`
**0xc0000142** is `STATUS_DLL_INIT_FAILED`

maybe something with manifest?

**Comment by SSE4** [comment]
as it happens only with MT builds, this explanation appears to be logical:
```
FlsAlloc returned FLS_OUT_OF_INDEXES, which in turned failed the CRT initialization. A little research shows that Fiber Local Storage has 128 slots per process; each DLL statically linked to CRT will take a slot during CRT initialization. There is possibility that if process has loaded too many DLLs and exhausted FLS slots, then new DLL statically linked to CRT will fail to load, and that is exactly what we ran into.
```
as GStreamer for sure has lots of DLLs, and we could have easily loaded 128 DLLs linked with static runtime.
I am not sure, is it worth to provide this configuration at all in this case (shared + static CRT?)

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**